src/main/java/org/jenkinsci/plugins/oic/OicAlgorithmValidator.java

```diff
@@ -0,0 +1,148 @@
+package org.jenkinsci.plugins.oic;
+
+import com.nimbusds.jose.Algorithm;
+import com.nimbusds.jose.EncryptionMethod;
+import com.nimbusds.jose.JWEAlgorithm;
+import com.nimbusds.jose.JWSAlgorithm;
+import com.nimbusds.jose.crypto.ECDSASigner;
+import com.nimbusds.jose.crypto.MACSigner;
+import com.nimbusds.jose.crypto.RSASSASigner;
+import com.nimbusds.jose.crypto.impl.AESCryptoProvider;
+import com.nimbusds.jose.crypto.impl.ContentCryptoProvider;
+import com.nimbusds.jose.crypto.impl.ECDHCryptoProvider;
+import com.nimbusds.jose.crypto.impl.PasswordBasedCryptoProvider;
+import com.nimbusds.jose.crypto.impl.RSACryptoProvider;
+import java.util.ArrayList;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.function.Function;
+import jenkins.security.FIPS140;
+
+/**
+ * This class helps in validating algorithms for FIPS compliance and filtering the non-compliant algorithms when in
+ * FIPS mode.
+ */
+public class OicAlgorithmValidator {
+
+    private static final boolean isFIPSMode = FIPS140.useCompliantAlgorithms();
+
+    /**
+     * Checks if the Jws signing algorithm used for OIC configuration is FIPS compliant.
+     */
+    public static boolean isJwsAlgorithmFipsNonCompliant(String algorithm) {
+        boolean matchNotFound = false;
+        if (isFIPSMode && algorithm != null) {
+            Set<JWSAlgorithm> jwsSupportedAlgorithms = new LinkedHashSet<>();
+            jwsSupportedAlgorithms.addAll(MACSigner.SUPPORTED_ALGORITHMS);
+            jwsSupportedAlgorithms.addAll(RSASSASigner.SUPPORTED_ALGORITHMS);
+            jwsSupportedAlgorithms.addAll(ECDSASigner.SUPPORTED_ALGORITHMS);
+
+            if (!jwsSupportedAlgorithms.isEmpty()) {
+                matchNotFound = jwsSupportedAlgorithms.stream()
+                        .map(JWSAlgorithm::getName)
+                        .noneMatch(name -> name.equals(algorithm));
+            }
+        }
+        return matchNotFound;
+    }
+
+    /**
+     *  Checks if the Jwe encryption algorithm used for OIC configuration is FIPS compliant.
+     */
+    public static boolean isJweAlgorithmFipsNonCompliant(String algorithm) {
+        boolean matchNotFound = false;
+        if (isFIPSMode && algorithm != null) {
+            Set<JWEAlgorithm> jweSupportedAlgorithms = new LinkedHashSet<>();
```

Comment:
same why not a static field with the collection already populated. I agree this will be helpful only if FIPS is activated :)

Reply:
Done